bridge/test/nftbridge/NFTBridge_test.js

```diff
@@ -826,6 +833,47 @@ contract("Bridge NFT", async function(accounts) {
         await assertTokenIsLockedInBridge(this.NFTBridge.address, anotherAccount, this.token);
       });
 
+      it("crosses an NFT back and forth emitting the expected events and leaving correct balances",
+          async function () {
+            // Side token is correctly claimed from side chain.
+            await assertTokenIsLockedInBridge(this.NFTBridge.address, anotherAccount, this.token);
+            await claimTokenFromBridgeEnsuringEventEmission(this.sideNFTBridge, anotherAccount, anAccount, tokenId,
+                tokenAddress, blockHash, transactionHash, logIndex);
+            let sideToken = await IERC721.at(sideTokenAddress);
+            await assertTokenHasBeenClaimed(this.sideNFTBridge.address, anotherAccount, sideToken);
+
+            // Cross back is started.
+            // Side chain bridge is allowed to move side token around.
+            let receipt = await sideToken.approve(this.sideNFTBridge.address, tokenId, {from: anotherAccount});
+            utils.checkRcpt(receipt);
+
+            // Side token is burned.
+            receipt = await this.sideNFTBridge.receiveTokensTo(
+                sideToken.address, anAccount, tokenId, {from: anotherAccount}
+            );
+            utils.checkRcpt(receipt);
+            await truffleAssert.fails(
+                sideToken.ownerOf(tokenId),
+                "ERC721: owner query for nonexistent token"
+            );
+
+            // Simulate Federator calling Bridge on main chain to accept the transfer.
+            const crossBackBlockHash = utils.getRandomHash();
+            const crossBackTransactionHash = utils.getRandomHash();
+            receipt = await this.NFTBridge.acceptTransfer(
+                tokenAddress, anotherAccount, anAccount, tokenId, crossBackBlockHash, crossBackTransactionHash,

```

Comment:
@pmprete is this correct? will the federator always call the bridges with the original main chain token address?

Reply:
yes, that's correct, the event emitted always has the original main chain token addreess